components/com_content/models/article.php

```diff
@@ -148,8 +148,21 @@ public function getItem($pk = null)
 				// Filter by published state.
 				$published = $this->getState('filter.published');
 				$archived = $this->getState('filter.archived');
+				
+				$queryy = $db->getQuery(true);
 
-				if (is_numeric($published))
+        			$queryy
+             				->select($db->quoteName('sharetoken'))
+             				->from($db->quoteName('#__share_draft'))
+             				->where($db->quoteName('articleId') . '=' . $pk);
```

Comment:
That is what I am saying, you don't need to do the token check if you use the token in the where clause.